apps/aehyperchains/test/aehc_chain_sim_connector.erl

```diff
@@ -15,80 +18,82 @@
 -export([handle_info/2]).
 -export([terminate/2]).
 
--export([send_tx/3, get_block_by_hash/1, get_top_block/0, dry_send_tx/3]).
-
-%% API.
-
--spec start_link(Args::map()) ->
-    {ok, pid()} | ingnore | {error, term()}.
-start_link(Args) ->
-    gen_server:start_link({local, ?MODULE}, ?MODULE, Args, []).
-
-stop() ->
-    gen_server:stop(?MODULE).
+-type block() :: aeconnector_block:block().
 
 %%%===================================================================
-%%%  aehc_connector behaviour
+%%%  aeconnector behaviour
 %%%===================================================================
 
--spec send_tx(binary(), binary(), binary()) -> binary().
-send_tx(Delegate, Commitment, PoGF) ->
-    gen_server:call(?MODULE, {send_tx, Delegate, Commitment, PoGF}).
+-spec connect(map(), function()) -> {ok, pid()} | {error, term()}.
+connect(Args, Callback) ->
+    Genesis = maps:get(<<"genesis_state">>, Args),
+    State = state(Genesis, Callback),
+    gen_server:start_link({local, ?MODULE}, ?MODULE, State, []).
+
+-spec dry_send_tx(binary(), binary()) -> boolean().
+dry_send_tx(Delegate, Payload) ->
+    gen_server:call(?MODULE, {dry_send_tx, Delegate, Payload}).
+
+-spec send_tx(binary(), binary()) -> ok | {error, term()}.
+send_tx(Delegate, Payload) ->
+    gen_server:call(?MODULE, {send_tx, Delegate, Payload}).
 
--spec get_top_block() -> aehc_parent_block:parent_block().
+-spec get_top_block() -> {ok, binary()} | {error, term()}.
 get_top_block() ->
     gen_server:call(?MODULE, {get_top_block}).
 
--spec get_block_by_hash(binary()) -> aehc_parent_block:parent_block().
+-spec get_block_by_hash(binary()) -> {ok, block()} | {error, term()}.
 get_block_by_hash(Hash) ->
     gen_server:call(?MODULE, {get_block_by_hash, Hash}).
 
--spec dry_send_tx(binary(), binary(), binary()) -> binary().
-dry_send_tx(Delegate, Commitment, PoGF) ->
-    gen_server:call(?MODULE, {dry_send_tx, Delegate, Commitment, PoGF}).
+-spec disconnect() -> ok.
+disconnect() ->
+    gen_server:stop(?MODULE).
 
 %%%===================================================================
 %%%  gen_server behaviour
 %%%===================================================================
 
--record(state, { pid::pid(), height = 0::non_neg_integer() }).
+-record(state, { pid::pid(), callback::function(), genesis::term() }).
+
+-type state() :: #state{}.
 
-init(Args) ->
+init(State) ->
```

Comment:
Shouldn't we use handle_continue here? Not a suggestion, just asking.

Reply:
We can remove agree (don't like redundant parts too)